tools/generator/Java.Interop.Tools.Generator.CodeGeneration/CodeGenerator.cs

```diff
@@ -804,15 +804,15 @@ public void WriteInterfaceInvoker (InterfaceGen @interface, string indent)
 			writer.WriteLine ();
 		}
 
-		public void WriteInterfaceListenerEvent (InterfaceGen @interface, string indent, string name, string nameSpec, string methodName, string full_delegate_name, bool needs_sender, string wrefSuffix, string add, string remove)
+		public void WriteInterfaceListenerEvent (InterfaceGen @interface, string indent, string name, string nameSpec, string methodName, string full_delegate_name, bool needs_sender, string wrefSuffix, string add, string remove, bool hasHandlerArgument = false)
 		{
 			writer.WriteLine ("{0}public event {1} {2} {{", indent, opt.GetOutputName (full_delegate_name), name);
 			writer.WriteLine ("{0}\tadd {{", indent);
 			writer.WriteLine ("{0}\t\tglobal::Java.Interop.EventHelper.AddEventHandler<{1}, {1}Implementor>(",
 					indent, opt.GetOutputName (@interface.FullName));
 			writer.WriteLine ("{0}\t\t\t\tref weak_implementor_{1},", indent, wrefSuffix);
 			writer.WriteLine ("{0}\t\t\t\t__Create{1}Implementor,", indent, @interface.Name);
-			writer.WriteLine ("{0}\t\t\t\t{1},", indent, add);
+			writer.WriteLine ("{0}\t\t\t\t{1},", indent, add + (hasHandlerArgument ? "_Helper": null));
```

Comment:
Please encode the event name into the "helper" method.  `Add_Helper` doesn't "feel" sufficiently unique to me in case a type has multiple events that require helpers.

Reply:
Thanks @jonpryor for your comment/suggestion.
The "add" is a variable/argument that already contains the name of the event as its content.
I have enhanced the name by adding _With_Handler_Helper in the end. so now we will have something like this:

```
public event EventHandler<Android.Media.AudioRoutingOnRoutingChangedEventArgs> RoutingChanged {
	add {
		global::Java.Interop.EventHelper.AddEventHandler<Android.Media.IAudioRoutingOnRoutingChangedListener, Android.Media.IAudioRoutingOnRoutingChangedListenerImplementor>(
				ref weak_implementor_AddOnRoutingChangedListener,
				__CreateIAudioRoutingOnRoutingChangedListenerImplementor,
				AddOnRoutingChangedListener_Helper_With_Handler_Helper,
				__h => __h.Handler += value);
	}
	remove {
		global::Java.Interop.EventHelper.RemoveEventHandler<Android.Media.IAudioRoutingOnRoutingChangedListener, Android.Media.IAudioRoutingOnRoutingChangedListenerImplementor>(
				ref weak_implementor_AddOnRoutingChangedListener,
				Android.Media.IAudioRoutingOnRoutingChangedListenerImplementor.__IsEmpty,
				__v => RemoveOnRoutingChangedListener (__v),
				__h => __h.Handler -= value);
	}
}

void AddOnRoutingChangedListener_Helper _With_Handler_Helper (Android.Media.IAudioRoutingOnRoutingChangedListener value)
{
	AddOnRoutingChangedListener (value, null);
}
```

Reply:
> The "add" is a variable/argument that already contains the name of the event as its content.

Then how/why does the unit test *not* have an event name?  https://github.com/xamarin/java.interop/pull/458/files#diff-be49ea1496d7371688060e0c36cccd7cR18

We're generating a `MyName` event, yet `MyName` doesn't appear in the method name `Add_Event_With_Handler_Helper()`.

Is the "problem" that we should specify `AddMyName` instead of `Add` here? https://github.com/xamarin/java.interop/pull/458/files#diff-c12b0a030f8bd272dcee2a5eca13e096R537

Reply:
@jonpryor the unit test is only passing "Add" as the event name, 

generator.WriteInterfaceListenerEvent (iface, string.Empty, "MyName", "MyNameSpec", "MyMethodName", "MyFullDelegateName", true, "MyWrefSuffix", "**Add**", "Remove", true);

it was a copy and paste from the previous test
https://github.com/xamarin/java.interop/blob/a9e6b5f61f4f04d57c9614145a7a7e50c7a91efd/tools/generator/Tests/Unit-Tests/CodeGeneratorTests.cs#L520


To make things more explicit/clear I have updated the unit test to pass "AddMyName"

generator.WriteInterfaceListenerEvent (iface, string.Empty, "MyName", "MyNameSpec", "MyMethodName", "MyFullDelegateName", true, "MyWrefSuffix", "**AddMyName**", "**RemoveMyName**", true);